packages/astro/src/content/content-layer.ts

```diff
@@ -230,6 +230,42 @@ export class ContentLayer {
 		const modulesImportsFile = new URL(MODULES_IMPORTS_FILE, this.#settings.dotAstroDir);
 		await this.#store.writeModuleImports(modulesImportsFile);
 		logger.info('Synced content');
+		if (this.#settings.config.experimental.contentIntellisense) {
+			await this.regenerateCollectionFileManifest();
+		}
+	}
+
+	async regenerateCollectionFileManifest() {
+		const collectionsManifest = new URL('collections/collections.json', this.#settings.dotAstroDir);
+		this.#logger.debug('content', 'Regenerating collection file manifest');
+		if (existsSync(collectionsManifest)) {
+			try {
+				const collections = await fs.readFile(collectionsManifest, 'utf-8');
+				const collectionsJson = JSON.parse(collections);
+				collectionsJson.entries ??= {};
+
+				for (const { hasSchema, name } of collectionsJson.collections) {
+					if (!hasSchema) {
+						continue;
+					}
+					const entries = this.#store.values(name);
+					if (!entries?.[0]?.filePath || entries[0].filePath.endsWith('.json')) {
```

Comment:
What does this `endsWith('.json')` condition prevents?

Reply:
The intellisense doesn't support JSON does it?

Reply:
You're correct (though there's no problems with them being in the manifest, tooling just ignore them), but I figure a whitelist would potentially be better here? Not sure if it's still the case in Content Layer, but YAML is supported inside data collections, for instance

Reply:
OK, I've removed it